alf/algorithms/on_policy_algorithm.py

```diff
@@ -66,14 +67,18 @@ def train_step(self, exp: Experience, state):
     def _train_iter_on_policy(self):
         """User may override this for their own training procedure."""
         alf.summary.get_global_counter().add_(1)
-        training_info = self.unroll(self._config.unroll_length)
-        training_info = training_info._replace(
-            rollout_info=(), info=training_info.rollout_info)
-        valid_masks = (training_info.step_type != StepType.LAST).to(
-            torch.float32)
-        loss_info, params = self.update_with_gradient(
-            self.calc_loss(training_info), valid_masks)
-        self.after_update(training_info)
-        self.summarize_train(training_info, loss_info, params)
-        self.summarize_metrics()
+
+        with record_time("time/unroll"):
+            training_info = self.unroll(self._config.unroll_length)
+
+        with record_time("time/train"):
+            training_info = training_info._replace(
+                rollout_info=(), info=training_info.rollout_info)
+            valid_masks = (training_info.step_type != StepType.LAST).to(
+                torch.float32)
+            loss_info, params = self.update_with_gradient(
+                self.calc_loss(training_info), valid_masks)
+            self.after_update(training_info)
+            self.summarize_train(training_info, loss_info, params)
```

Comment:
while you are at it, let's also separately log summary time?

Reply:
summary is distributed in lot of places. It's not easy to collect them all.